File: src/lib/types/color_palette.ts
export const DEFAULT_PALETTE_ID = 'palette-vivid';
export const FALLBACK_ISSUE_COLOR = '#ef4444';

export interface ColorPalette {
	id: string;
	name: string;
	/** Hex color strings, e.g. ["#ef4444", "#f97316"] */
	colors: string[];
	is_built_in: boolean;
}

export interface CreateColorPaletteRequest {
	name: string;
	colors: string[];
}

export interface UpdateColorPaletteRequest {
	id: string;
	name?: string;
	colors?: string[];
}

File: src-tauri/src/models/color_palette.rs
use serde::{Deserialize, Serialize};
use ts_rs::TS;

#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[ts(export)]
pub struct ColorPalette {
    pub id: String,
    pub name: String,
    /// JSON array of hex color strings, e.g. ["#ef4444", "#f97316"]
    pub colors: Vec<String>,
    pub is_built_in: bool,
}

#[derive(Debug, Deserialize)]
pub struct CreateColorPaletteRequest {
    pub name: String,
    pub colors: Vec<String>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateColorPaletteRequest {
    pub id: String,
    pub name: Option<String>,
    pub colors: Option<Vec<String>>,
}
